Bump maven from 3.8.1 to 3.8.4

jorsol opened the pull request.

jorsol committed:
Bump maven from 3.8.1 to 3.8.4

Signed-off-by: Jorge Solórzano <jorsol@gmail.com>

famod commented:
Thanks! Can you please rebase on `main` to receive the latest CI fixes?
Those scala tests are "new" though, so you might need to take a closer look at those.

jorsol commented:
> Thanks! Can you please rebase on `main` to receive the latest CI fixes? Those scala tests are "new" though, so you might need to take a closer look at those.

Rebased!

jorsol commented:
Not sure if the errors of Scala on **Windows** are related to this PR.

famod committed:
Update scala-maven-plugin to 4.5.4 to fix jansi incompatibility with Maven 3.8.4

Also make sure the plugin is present in local repo _before_ the ITs are executed.

famod commented:
Let's see what CI has to say to the commit I've just pushed.

famod commented:
Looking much better now:
- `integration-tests/grpc-hibernate` fails with heap space issues while building the jar (FYI @gsmet, the same we saw in `docs`)
- `integration-tests`  still has one test failure but that's due to a timeout because it seems the plugin download took too long

I will have another look later today.

jorsol commented:
> Looking much better now

Nice, thanks for checking this out, I also will have another look later today.

famod commented:
As for the failure in `integration-tests/scala`, I think it just takes too long to download the (partially outdated) plugins, especially the `scala-maven-plugin`.
Sure, we could simply increase the timeout, but the actual problem lies deeper. I'll try to have the plugin downloaded by the "outer" build process, so that it ends up in the Maven repo cache (in the initial build) and I'll also try to align surefire version.

The heap issue doesn't seem related.

famod commented on independent-projects/bootstrap/maven-resolver/pom.xml:
```diff
@@ -56,6 +56,24 @@
+                </exclusion>
+                <exclusion>
+                    <groupId>javax.enterprise</groupId>
+                    <artifactId>cdi-api</artifactId>
```
I need to take a closer look at this, given that it should have been removed in 3.8.3: https://issues.apache.org/jira/browse/MNG-7045

Also, I think the remaining exclusions need to be defined in the bootstrap BOM, not here...?

famod replied:
As I thought: not needed anymore!

I have added two more commits locally that I will give a spin in fork's CI. If all goes to plan I will update this branch/PR.

@jorsol I hope you don't mind if I amend your first commit?

jorsol replied:
Sure @famod, I don't mind if you amend my commit.

famod replied:
Ok thanks, done!

famod committed:
Clean up more Maven bootstrap related exclusions

famod reviewed (approved):
Should be complete now!

@ia3andy @aloubyansky is there anything else that needs to be adjusted so that codestarts use Maven 3.8.4?

aloubyansky commented on independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/SecDispatcherImpl.java:
```diff
@@ -44,11 +44,7 @@
-        } catch (PlexusCipherException e) {
-            throw new IllegalStateException("Failed to init Security Dispatcher", e);
-        }
+        this._cipher = new DefaultPlexusCipher();
```
Could we still keep try/catch catching a `Throwable` or something to stay backward compatible?

jorsol replied:
Backward compatible? this actually was catching a specific checked exception `PlexusCipherException` and wrapped it in a runtime exception `IllegalStateException`, since the instantiation of `DefaultPlexusCipher` doesn't throw the checked exception anymore I don't see why trying to catch "all" exceptions (`Throwable`) would make this backward compatible.

famod replied:
FTR:
- https://github.com/codehaus-plexus/plexus-cipher/commit/48b5a36bbace09d12f3f9ee24d516c88dd9a5e10#diff-edc9e74df359f6207f332996ff040bfe9bbea1f9e7800b71a0cae5f0c6346f7bL41
- https://github.com/codehaus-plexus/plexus-cipher/commit/48b5a36bbace09d12f3f9ee24d516c88dd9a5e10#diff-e34d3c221117a8ce5c9b679f4acdd7dede829586c9bc5cb907e099d9eb61aa96L77

aloubyansky replied:
I quickly tried to break it but couldn't, so it's ok.

gastaldi replied:
According to @famod links above, I think it could break only if the `SHA-256` algorithm weren't available, which isn't the case since this algorithm is guaranteed to be supported in a JCA-compliant JVM

aloubyansky commented on independent-projects/bootstrap/pom.xml:
```diff
@@ -33,11 +33,11 @@
+        <maven-shared-utils.version>3.3.4</maven-shared-utils.version> <!-- changed from the original version due to https://snyk.io/vuln/maven:org.apache.maven.shared:maven-shared-utils@3.2.1 -->
         <maven-wagon.version>3.4.3</maven-wagon.version>
         <httpcore.version>4.4.14</httpcore.version><!-- Keep in sync with maven-wagon.version (wagon-http 3.4.3 brings in 4.4.13 and 4.4.14) -->
         <plexus-cypher.version>1.7</plexus-cypher.version>
```
BTW, this property doesn't seem to be used, afaics.

jorsol replied:
I will check it and remove it if needed, thanks!

aloubyansky reviewed (approved):
@gsmet wdyt?

gastaldi reviewed (approved):
LGTM

gsmet commented:
Nice work, everyone. Once this is in, I will upgrade the Maven wrapper used in the platform.

famod commented:
@gsmet
> Nice work, everyone. Once this is in, I will upgrade the Maven wrapper used in the platform.

Are you also going to update the quickstart ones?

famod commented:
@gastaldi did you re-trigger CI? That general failure in `Native Tests - Data3` was just a random blurb. It was fine before... 🤷‍♂️

gastaldi commented:
@famod yes, I did, I couldn't read the logs so I retriggered the build. Thanks for the heads-up! I guess it's fine to merge this when it's done
